### ISSUE TITLE
Feature - Private Mode for Images Only

### ISSUE THREAD

**Issue by VirusThePanda** [afollestad/impression#293]
Thought this could be a good feature to put in. But maybe we could put in a hidden mode or private mode for photos too? Sort of like how we can exclude albums.

**Comment by afollestad** [comment]
We could use **Digitus** for authentication, also. But we'd need to encrypt the photos so apps outside of Impression wouldn't be able to see them. https://github.com/afollestad/digitus

**Monospace** by Jack Underwood is a good example of this; you can encrypt notes and open/un-encrypt them using the fingerprint dialog.

**Comment by VirusThePanda** [comment]
Is Monospace Open Source? I could possibly look at the code and start
working on this. (Once I can that is. Swamped with school work).

**Comment by plusCubed** [comment]
Might be helpful: https://github.com/charlesconnell/ImageHider

**Comment by afollestad** [comment]
I don't think so, I might add some encryption features to Digitus at some point since that's actually included with this APIs used for fingerprint authentication. 

But yeah, probably not for weeks, I work 8+ hours everyday doing dev for work so I don't have a lot of time for any personal projects.

**Comment by VirusThePanda** [comment]
Same here about the time. Mostly because of school here. But I will look into it as soon as I can. Might be able to play around with Digitus in the emulator or something and I can possibly figure it out then. Or a possible work around until encryption, we could make a hidden folder that normally, a gallery can't see. Kinda like how music albums won't pop up in a gallery most times.

**Comment by alexsmithfanning** [comment]
A great workaround I saw a long time ago:

Normally, when you store images on your phone, it goes in its own dedicated pictures folder. But, if its *moved* to the directory where Impression (I believe that's /storage/emulated/0/Android/data) stores its settings, then no other app can access it. Not even the user in normal unrooted circumstances. I don't know if that's the case now, but it could work.

**Comment by plusCubed** [comment]
@alexsmithfanning Good idea! It is definitely still the case now and with such a solution it would cover most users who aren't rooted (and for rooted users, they would be able to grab the key in an encryption solution anyway). Clearing data or uninstalling the app would erase the secured images though, but I suppose with encryption the key would get erased too.

**Comment by alexsmithfanning** [comment]
Not true actually. I think there's an option you can set in (pulling things directly from my ass here, don't kill me) a flag, a variable, or something Guava related that I don't know about because I'm not a Java programmer.

I still have app data in those directories from apps that I uninstalled months ago. Resources, images, all of it.

**Do note that I am not rooted and I can only view those resources because I extracted them from my SD Card**.

**Comment by plusCubed** [comment]
@alexsmithfanning External storage SD cards are viewable by all apps and users, and some apps store stuff there. Internal storage, in /data/data, apps have access to their own directories and you can't access them without root (if you're familiar with linux/unix, each app is basically a user). Their folders are deleted when uninstalling (otherwise that'd be inaccessible data if the app is never installed again, just taking up space).

**Comment by alexsmithfanning** [comment]
Ah, okay. That does make sense.

You learn something new everyday.
